### PR TITLE
Simplify setting the working directory

### LINKED ISSUE #550: Changing directory should be visible and set `default-directory`
Not visible in the inferior:

* `ess-use-dir`

Doesn't update `default-directory`:

* `ess-use-dir`
* `ess-set-working-directory`

### PR DESCRIPTION
**Pull request by jabranham**
alias ess-use-dir to ess-set-working-directory since they do
essentially the same thing. Move ess-use-this-dir to ess-inf since it
can be used by non-S/R languages now.

Closes #550

### CODE REVIEW

**Comment by jabranham** [comment]
It would make more sense (to me) if `ess-use-this-dir` set the working directory to where the file is located, not where `default-directory` is. OK to make that change?

**Comment by vspinu** [comment]
That makes sense. But it should fall back on default directory, something like

```el
(if buffer-file-name
      (file-name-directory buffer-file-name)
    default-directory)
```

**Comment by jabranham** [comment]
Thanks, done

File: lisp/ess-inf.el [hunk 1]
```diff
@@ -3073,6 +3074,20 @@ subprocess and Emacs buffer `default-directory'."
       (error "Not implemented for dialect %s" ess-dialect))))
 
 (defalias 'ess-change-directory 'ess-set-working-directory)
+(define-obsolete-function-alias
+  'ess-use-dir 'ess-set-working-directory "2018-06-11")
+
+(defun ess-use-this-dir (&optional no-force-current)
+  "Set the current process directory to the directory of this file.
+
```
Comment (lionel-) [inline lisp/ess-inf.el]: Small nitpick: there shouldn't be a newline after the leading sentence.

Comment (jabranham) [reply lisp/ess-inf.el]: Removed

**Comment by mmaechler** [comment]
Hmm, but we already have the functionality of  your `ess-use-this-dir` -- even with the exact same name in `ess-s-lang.el` .. and I have been using it every day via `M-[Enter]` ... ???

**Comment by jabranham** [comment]
@mmaechler Right, this moves the function into `ess-mode.el` since it can now be used by languages other than R/S. It doesn't change the `M-RET` binding in S mode, so it'll still be there.

**Comment by mmaechler** [comment]
aah.. now I understand.  Ok.

**Comment by jabranham** [comment]
@lionel-  I'm trying to add a test:

```emacs-lisp
(ert-deftest ess-set-working-directory ()
  (with-r-running nil
    (ess-set-working-directory "/")
    (insert "getwd()\n")
    (should (output= (ess-eval-buffer nil)
              "setwd('/')\n>\n> getwd()\n[1] \"/\""))))
```

But it's failing because the output comes back with fontification stuff:

```
F ess-set-working-directory
    (ert-test-failed "Expected: 
setwd('/')
>
> getwd()
[1] \"/\""
                     #("Result: 
setwd('/')
> 
> getwd()
[1] \"/\"" 9 19
(fontified nil font-lock-face comint-highlight-input front-sticky t field input)
19 20
(fontified nil font-lock-face comint-highlight-input)
20 22
(fontified nil)
22 33
(fontified nil)
33 40
(fontified nil)))
```

Is there a better way to do this?

**Comment by jabranham** [comment]
Thanks. In the meantime, I pushed a test that tests `default-directory` gets changed.

**Comment by lionel-** [comment]
Also you should send a command to the inferior rather than insert text in the buffer.

**Comment by jabranham** [comment]
Thanks, that seems to make the test work, for me at least. We'll wait to see if travis complains.

**Comment by lionel-** [comment]
Great, one more test towards good coverage!
Thanks for the PR.

**Pull request merged**